### PR TITLE
TestMultipleSchedulers.test_fairshare_usage fails due to race condition

### PR DESCRIPTION
**Pull request by Bhagat-Rajput**
#### Bug/feature Description
* **TestMultipleSchedulers.test_fairshare_usage** fails sometimes due to race condition.
#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* In testcase PBS can run only 1 job at a time(because assigned 1 ncpus only) and it can't run the next job until previous one finished it's execution.

#### Solution Description
* Added code to delete the job once we are done with the job means will not wait for the job to be finished.

#### Testing logs/output
* Testing log: [ptl_test_fairshare_usage.txt](https://github.com/PBSPro/pbspro/files/2763727/ptl_test_fairshare_usage.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_multi_sched.py [hunk 1]
```diff
@@ -700,16 +699,23 @@ def test_fairshare_usage(self):
         self.server.expect(JOB, {'job_state': 'Q'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: When you will delete sc1_jid1, a sched cycle will be kicked automatically. So, kicking another one here is redundant now, so please remove this.

File: test/tests/functional/pbs_multi_sched.py [hunk 2]
```diff
@@ -700,16 +699,23 @@ def test_fairshare_usage(self):
...
                             id='sc1')
+        # need to delete the running job because PBS has only 1 ncpu and
+        # our work is also done with the job.
+        # this step will decrease the execution time as well
+        self.server.delete(sc1_jid1, wait=True)
         # pbsuser3 job will run after pbsuser1
         self.server.expect(JOB, {'job_state': 'R'}, id=sc1_jid3)
         self.server.expect(JOB, {'job_state': 'Q'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: Same comment here

File: test/tests/functional/pbs_multi_sched.py [hunk 3]
```diff
@@ -700,16 +699,23 @@ def test_fairshare_usage(self):
...
                             id='sc1')
+        # deleting the currently running job
+        self.server.delete(sc1_jid3, wait=True)
         # pbsuser2 job will run in the end
         self.server.expect(JOB, {'job_state': 'R'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: same comment here

File: test/tests/functional/pbs_multi_sched.py [hunk 4]
```diff
@@ -738,16 +744,21 @@ def test_fairshare_usage(self):
         self.server.expect(JOB, {'job_state': 'Q'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: same comment here

File: test/tests/functional/pbs_multi_sched.py [hunk 5]
```diff
@@ -738,16 +744,21 @@ def test_fairshare_usage(self):
...
                             id='sc1')
+        # deleting the currently running job
+        self.server.delete(sc1_jid4, wait=True)
         # pbsuser1 job will run after pbsuser4
         self.server.expect(JOB, {'job_state': 'R'}, id=sc1_jid1)
         self.server.expect(JOB, {'job_state': 'Q'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: same comment here

File: test/tests/functional/pbs_multi_sched.py [hunk 6]
```diff
@@ -738,16 +744,21 @@ def test_fairshare_usage(self):
...
                             id='sc1')
+        # deleting the currently running job
+        self.server.delete(sc1_jid1, wait=True)
         # pbsuser2 job will run in the end
         self.server.expect(JOB, {'job_state': 'R'}, id=sc1_jid2)
         self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_multi_sched.py]: same comment here

File: test/tests/functional/pbs_multi_sched.py (hunk 1 above)
Comment (Bhagat-Rajput) [reply test/tests/functional/pbs_multi_sched.py]: Thanks @agrawalravi90 for the review, i tried your solution but couldn't get the expected output.
As per my understanding, need to trigger the scheduling cycle when job should be in 'R' state to update the fairshare usage file correctly.

Comment (agrawalravi90) [reply test/tests/functional/pbs_multi_sched.py]: ah, ok, i think you are right, the fairshare usage won't get updated for sc1_jid1 's user as we will deleted the job and then kick the sched cycle. My bad, I take my comments back.

**Comment by agrawalravi90** [comment]
@Bhagat-Rajput please rebase in preparation for merge

**Comment by Bhagat-Rajput** [comment]
Thanks @agrawalravi90 and @anamikau for the sign-off. Rebased and ready to merge.

**Comment by mike0042** [comment]
Please re-sign your commit to initiate a new AppVeyor build @Bhagat-Rajput

**Comment by Bhagat-Rajput** [comment]
Thanks @mike0042 ,I have rebased and all checks are passed now so please take it furthur.

**Comment by agrawalravi90** [comment]
@Bhagat-Rajput your commit is still not signed. Please sign it and commit again. Also, for future, please turn on git autosign so that you don't face delays like this.

**Comment by Bhagat-Rajput** [comment]
Thanks @agrawalravi90 for a valuable suggestion. I have signed the commit as well as configured the git as well.

**Pull request merged**